cli/glnexus_cli.cc

```diff
@@ -743,21 +760,37 @@ int main_unify_sites(int argc, char *argv[]) {
     }
 
     vector<pair<string,size_t> > contigs;
-    vector<GLnexus::range> ranges;
     vector<GLnexus::discovered_alleles> valleles;
     H("merge discovered allele files",
-      utils::merge_discovered_allele_files(console, discovered_allele_files, contigs, ranges, valleles));
+      utils::merge_discovered_allele_files(console, discovered_allele_files, contigs, valleles));
 
     unsigned discovered_allele_count=0;
     for (const auto& dsals : valleles) {
         discovered_allele_count += dsals.size();
     }
     console->info() << "consolidated to " << discovered_allele_count << " alleles for site unification";
 
+    set<GLnexus::range> ranges;
+    if (!bedfilename.empty()) {
+        string range_txt; // empty string, the plan is to get rid of the argument
+        vector<GLnexus::range> i_ranges;
+        H("Parsing range argument", parse_ranges_from_cmd_line(bedfilename, range_txt, contigs, i_ranges));
+        for (auto &r : i_ranges)
+            ranges.insert(r);
+    }
+
     vector<GLnexus::unified_site> sites;
-    for (int i = 0; i < valleles.size(); i++) {
+    for (const auto& dsals : valleles) {
+        if (dsals.size() == 0) continue;
+
+        // find the containing range
+        const GLnexus::range &pos = dsals.begin()->first.pos;
+        GLnexus::range containing_range(-1,-1,-1);
+        H("find containing range", utils::find_containing_range(ranges, pos, containing_range));
+
         vector<GLnexus::unified_site> sites_i;
-        H("unify sites", GLnexus::unified_sites(unifier_cfg, valleles[i], sites_i, ranges[i]));
+        H("unify sites",
+          GLnexus::unified_sites(unifier_cfg, dsals, sites_i, containing_range));
```

Comment:
When we have just one `discovered_alleles` structure, we won't immediately know the containing range here. We'll only be able to find it for each site after `unified_sites` has defined them. `unified_sites` could take in the list (hopefully interval tree) of containing target ranges, and add them. Or we could do a second pass over the sites after running `unified_sites` to add the containing target ranges.